src/sagas/walletConnect.js

```diff
@@ -0,0 +1,424 @@
+
+  const activeSessions = yield call(() => web3wallet.getActiveSessions());
+  for (const key of Object.keys(activeSessions)) {
+    yield call(() => web3wallet.disconnectSession({
+      topic: activeSessions[key].topic, 
+      reason: {
+        code: USER_DISCONNECTED,
+        message: '',
+      },
+    }));
+  }
+
+  yield call(refreshActiveSessions);
+}
+
+export function* onSessionRequest(action) {
+  const { payload } = action;
+  const { params } = payload;
+
+  const { web3wallet } = yield select((state) => state.walletConnect);
+  const activeSessions = yield call(() => web3wallet.getActiveSessions());
+  const requestSession = activeSessions[payload.topic];
+  if (!requestSession) {
+    console.error('Could not identify the request session, ignoring request..');
+    return;
+  }
+
+  const data = {
+    icon: get(requestSession.peer, 'metadata.icons[0]', null),
+    proposer: get(requestSession.peer, 'metadata.name', ''),
+    url: get(requestSession.peer, 'metadata.url', ''),
+    description: get(requestSession.peer, 'metadata.description', ''),
+  };
+
+  switch(params.request.method) {
+    case 'hathor_signMessage':
+      yield put({
+        type: 'SIGN_MESSAGE_REQUEST',
+        payload: {
+          ...data,
+          requestId: payload.id,
+          topic: payload.topic,
+          message: get(params, 'request.params.message'),
+        }
+      });
+    break;
+    default:
+      yield call(() => web3wallet.respondSessionRequest({
+        topic: payload.topic,
+        response: {
+          id: payload.id,
+          jsonrpc: '2.0',
+          error: {
+            code: USER_REJECTED_METHOD,
+            message: 'Rejected by the user',
+          },
+        },
+      }));
+    break;
+  }
+}
+
+export function* onSignMessageRequest(action) {
+  const data = action.payload;
+  const { web3wallet } = yield select((state) => state.walletConnect);
+
+  const onAcceptAction = { type: 'WALLET_CONNECT_ACCEPT' };
+  const onRejectAction = { type: 'WALLET_CONNECT_REJECT' };
+
+  yield put(setWalletConnectModal({
+    show: true,
+    type: WalletConnectModalTypes.SIGN_MESSAGE,
+    data,
+    onAcceptAction,
+    onRejectAction,
+  }));
+
+  const { accept } = yield race({
+    accept: take(onAcceptAction.type),
+    reject: take(onRejectAction.type),
+  });
+
+  try {
+    if (!accept) {
+      yield call(() => web3wallet.respondSessionRequest({
+        topic: payload.topic,
+        response: {
+          id: payload.id,
+          jsonrpc: '2.0',
+          error: {
+            code: USER_REJECTED,
+            message: 'Rejected by the user',
+          },
+        },
+      }));
+      return;
+    }
+
+    const wallet = yield select((state) => state.wallet);
+
+    if (!wallet.isReady()) {
+      console.error('Got a session request but wallet is not ready, ignoring..');
+      return;
+    }
+
+    const signedMessage = yield call(() => wallet.signArbitraryMessage(data.message));
```

Comment:
Do we have this method in the wallet?

Reply:
We don't have it yet, I'll open a PR to create it.